link/github.go

```diff
@@ -34,7 +34,7 @@ func (s *github) ValidateURL(name string, u *url.URL) (*url.URL, error) {
 	if len(paths) != 2 {
 		return nil, errors.Errorf("path invalid %s for url %s", paths, u)
 	}
-	if paths[0] != name {
+	if strings.ToLower(paths[0]) != name {
```

Comment:
I think maybe we should normalize the URL name to be all lowercase. That way it's lowercase in the user statement just like the user name.

Reply:
That sounds pretty clean, but I am starting to wonder something now.

When I look at `user/user.go`, I see there is a normalizeURL function which does nothing except parse and then output the string again. The `User` struct contains a string for URL, but the service itself parses it as an *url.URL.

Would it make sense to have the `User` struct store the URL field as an `*url.URL` and have every service provide a `parseAndNormalizeURL` function, which takes the string and returns a normalized `*url.URL`? That would make it more convenient to normalize the URL before validating it. Or is there a reason you do not want to store the URL as an `*url.URL` object in the `User` struct? I'm fairly new to Go so there may be valid reasons I'm totally missing.

Reply:
Yeah *url.URL has a lot of extra stuff (port, query, anchor, username@password, etc) and I'm thinking since we want the url to be in a canonical/normalized form using a string I think is a good thing.

Maybe even don't parse it to an URL until the very last moment.

And maybe we should restrict urls to only include scheme://domain/path and nothing else is allowed, just to be safe.

Reply:
Okay, then I think I'll let the normalizeURL function take a string, strip out all that stuff we don't want/need, lowercase it and returns that as string.